### DISCUSSION TITLE
Issues working with the Compute Shader Functionality -- Basic Example

### DISCUSSION THREAD

**Discussion by FranciscoJintoFox** [raysan5/raylib#2734] [Q&A]
Hello, 
I'm having trouble getting a compute shader to work with any functionality. My goal is to pass in an image (or image data), and pass out an output image generated pixel by pixel that I can render to the screen; ideally with the built-in DrawTexture(). 
I have simplified my code in an attempt to output a single float, written as a constant in the compute shader. I'm also new to Shaders so I may have missed something--but I can't tell what.

Excerpts:
`[...]`
`//Qualify internal format of the (previously) set up output image`
`layout(std430, binding = 0) buffer rbcLayout{`
`   float rbcBufferDest;`
`};`

> 

`void main(){`
`   [...]`
`   rbcBufferDest = 0.25;`
`}`
In the main file, I'm trying to use `rlReadShaderBufferElements(ssboA, &temp, sizeof(float), 0);` to read the data back into a float. However, printing this value returns either -0.000000 or -nan. I've been trawling through various documentation and tutorials/blogs all day to no avail.

I believe my Raylib and openGL versions are up to date (4.2 and 4.6 respectively).

Thank you to anyone who can give me some advice. 

Oh right my compile command is:
`gcc -std=c++11 wave8.cpp -lraylib -lGL -lm -lpthread -lrt -lX11 -o wave8.run`

I read that hardware could cause issues--I'm running a Radeon RX480 with Ryzen 5 2600X, on an x470 motherboard. 

[rbc.glsl.txt](https://github.com/raysan5/raylib/files/9689433/rbc.glsl.txt)
[rbc_render.glsl.txt](https://github.com/raysan5/raylib/files/9689434/rbc_render.glsl.txt)
[terminalOutput.txt](https://github.com/raysan5/raylib/files/9689435/terminalOutput.txt)
[wave8.cpp.txt](https://github.com/raysan5/raylib/files/9689436/wave8.cpp.txt)

**Comment by FranciscoJintoFox** [comment]
So I spent some time digging through rlgl.h, the Khronos docs, and some blogs and was able to put together a basic example using (mostly) the raw GL functions. Not sure why this works but the rlgl (which is almost identical) doesn't.
[shadertest.cpp.txt](https://github.com/raysan5/raylib/files/9694235/shadertest.cpp.txt)
[test.glsl.txt](https://github.com/raysan5/raylib/files/9694236/test.glsl.txt)
[output.txt](https://github.com/raysan5/raylib/files/9694244/output.txt)

**Comment by ckw** [comment]
Make sure your raylib is compiled with support for opengl 4.3. See https://github.com/raysan5/raylib/issues/2911#issuecomment-1571181279

**Comment by ElectroidDes** [comment]
Don't use raylib - use godot. There are fully working compute shaders. Why do you need this raylib?

**Reply by jamesl-github** [reply]
don't use godot -- use unreal --- godot is a toy compared to unreal -- no real AAA uses godot

**Reply by TonyC-code** [reply]
> Don't use raylib - use godot. There are fully working compute shaders. Why do you need this raylib?

Shut up.. why are you here. Go away.

**Reply by Egyember** [reply]
real programmers use assembly and self modifying code
